src/internet_identity/src/state/flow_states.rs

```diff
@@ -0,0 +1,131 @@
+use crate::anchor_management::registration::Base64;
+use crate::MINUTE_NS;
+use candid::Principal;
+use ic_cdk::api::time;
+use internet_identity_interface::internet_identity::types::Timestamp;
+use std::collections::{HashMap, VecDeque};
+
+// Expiration for a registration flow, the same as the captcha expiry
+const FLOW_EXPIRATION_NS: u64 = 5 * MINUTE_NS;
+
+#[derive(Default)]
+pub struct FlowStates {
+    /// In-progress registration flows tied to the respective principal. The principal will become
+    /// a temp key upon successful completion of the registration flow.
+    /// Registration flows are removed on three conditions:
+    /// * The registration flow is completed successfully
+    /// * Once the flow expires
+    /// * The Internet Identity canister is upgraded
+    ///
+    /// The total number of flow states is limited by the registration rate limit.
```

Comment:
Given the captcha is now tied to the flow_state and the number of flow_states is limited by the registration rate limit, the `max_unsolved_captchas` config option in the captcha config only has an effect on the legacy registration flow.

Reply:
Doesn't that depend on what comes furst, the registration rate limit or the `max_unsolved_captchas`?

Reply:
No, `max_unsolved_captchas` is not used at all in the context of the V2 registration flow, as we store the captcha with the flow state.